x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/geo_grid.test.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act } from 'react-dom/test-utils';
+import { setup, SetupResult, getProcessorValue, setupEnvironment } from './processor.helpers';
+
+const GEO_GRID_TYPE = 'geo_grid';
+
+describe('Processor: GeoGrid', () => {
+  let onUpdate: jest.Mock;
+  let testBed: SetupResult;
+  const { httpSetup } = setupEnvironment();
+
+  beforeAll(() => {
+    jest.useFakeTimers({ legacyFakeTimers: true });
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  beforeEach(async () => {
+    onUpdate = jest.fn();
+
+    await act(async () => {
+      testBed = await setup(httpSetup, {
+        value: {
+          processors: [],
+        },
+        onFlyoutOpen: jest.fn(),
+        onUpdate,
+      });
+    });
+
+    const { component, actions } = testBed;
+
+    component.update();
+
+    // Open flyout to add new processor
+    actions.addProcessor();
+    // Add type (the other fields are not visible until a type is selected)
+    await actions.addProcessorType(GEO_GRID_TYPE);
+  });
+
+  test('prevents form submission if required fields are not provided', async () => {
+    const {
+      actions: { saveNewProcessor },
+      form,
+    } = testBed;
+
+    // Click submit button with only the type defined
+    await saveNewProcessor();
+
+    // Expect form error as "field" is a required parameter
+    expect(form.getErrorsMessages()).toEqual([
+      'A field value is required.', // "Field" input
+      'A tile type value is required.', // "Tile type" input
+    ]);
+  });
+
+  test('saves with default parameter values', async () => {
+    const {
+      actions: { saveNewProcessor },
+      form,
+    } = testBed;
+
+    // Add "field" value
+    form.setInputValue('fieldNameField.input', 'test_geo_grid_processor');
+
+    // Add "tile tyle" field
+    form.setSelectValue('tileTypeField', 'geohex');
+
+    // Save the field
+    await saveNewProcessor();
+
+    const processors = getProcessorValue(onUpdate, GEO_GRID_TYPE);
+
+    expect(processors[0][GEO_GRID_TYPE]).toEqual({
+      field: 'test_geo_grid_processor',
+      tile_type: 'geohex',
+    });
+  });
+
+  test('saves with optional parameter values', async () => {
```

Comment:
Is it possible to add a test case where the pipeline is actually created?

It seems that when you add a processor with wrong field names (e.g `parent` instead of `parent_field`) to a pipeline you don't see any error messages until you click the `Save pipeline` button, so it might be useful to have a test which creates a pipeline with a processor where all fields are filled in so that it can catch any misspelled field names. This might also be helpful in the future if any of the processor field names is updated from the Es side, this test will fail and so it will indicate that any of the field names has been changed and so we need to change it on the UI side as well.

Reply:
I like this idea, although it might need to be a functional test instead of a CIT to verify them against ES since with jest we always use mocks to compare with the UI input. I'll create a separate issue for exploring this further!